whirlpool/appliance.py

```diff
@@ -51,12 +51,13 @@ def unregister_attr_callback(self, update_callback: Callable):
             LOGGER.error("Attr callback not found")
 
     def _event_socket_handler(self, msg):
-        json_msg = json.loads(msg)
-        timestamp = json_msg["timestamp"]
-        for attr, val in json_msg["attributeMap"].items():
-            if not self.has_attribute(attr):
-                continue
-            self._set_attribute(attr, str(val), timestamp)
+        if msg is not None:
+            json_msg = json.loads(msg)
+            timestamp = json_msg["timestamp"]
+            for attr, val in json_msg["attributeMap"].items():
+                if not self.has_attribute(attr):
+                    continue
```

Comment:
This was a mistake from the beginning - we need to tell HA to update (via calback) after we sync the data.  The base code is only updating the internal data.

Reply:
I since moved this around a bit - rather than calling the update with (None), lets just update HA whenever a fetch occurs.  That way it just "works" whenever a fetch is triggered.